src/scripts/dimApp.config.js

```diff
@@ -1,194 +1,170 @@
-(function() {
-  'use strict';
-
-  console.time('First item directive built');
-
-  angular.module('dimApp')
-    .config((localStorageServiceProvider) => {
-      localStorageServiceProvider.setPrefix('');
-    });
+console.time('First item directive built');
+
+const angular = require('angular');
+
+angular.module('dimApp')
+  .config((localStorageServiceProvider) => {
+    localStorageServiceProvider.setPrefix('');
+  });
+
+// Putting this comparison in a function defeats a constant-folding optimization
+function compare(a, b) {
+  return a === b;
+}
+
+angular.module('dimApp')
+  .value('dimPlatformIds', {
+    xbl: null,
+    psn: null
+  })
+  .value('dimState', {
+    membershipType: -1,
+    active: null,
+    debug: false
+  })
+  .value('dimFeatureFlags', {
+    isExtension: window.chrome && window.chrome.extension,
+    // Tags are off in release right now
+    tagsEnabled: !compare('$DIM_FLAVOR', 'release'),
+    compareEnabled: true,
+    vendorsEnabled: true,
+    qualityEnabled: true,
+    // Additional debugging / item info tools
+    debugMode: false,
+    // Print debug info to console about item moves
+    debugMoves: false,
+    // show changelog toaster
+    changelogToaster: compare('$DIM_FLAVOR', 'release') || compare('$DIM_FLAVOR', 'beta'),
+
+    materialsExchangeEnabled: !compare('$DIM_FLAVOR', 'release'),
+    // allow importing and exporting your DIM data to JSON
+    importExport: compare('$DIM_FLAVOR', 'release')
+  })
+  .factory('loadingTracker', function(promiseTracker) {
+    return promiseTracker();
+  });
+
+
+angular.module('dimApp')
+  .run(function($window,
+                $rootScope,
+                $translate,
+                loadingTracker,
+                SyncService,
+                dimInfoService,
+                dimFeatureFlags,
+                dimSettingsService) {
+    $rootScope.loadingTracker = loadingTracker;
+
+    // 1 Hour
+    $rootScope.inactivityLength = 60 * 60 * 1000;
+
+    $rootScope.isUserInactive = function() {
+      var currentTime = Date.now();
+
+      // Has This User Been Inactive For More Than An Hour
+      return ((currentTime) - $rootScope.lastActivity) > $rootScope.inactivityLength;
+    };
+
+    $rootScope.trackActivity = function() {
+      $rootScope.lastActivity = Date.now();
+    };
+
+    // Track Our Initial Activity of Starting the App
+    $rootScope.trackActivity();
+
+    $window.initgapi = function() {
+      SyncService.init();
+    };
+
+    var chromeVersion = /Chrome\/(\d+)/.exec($window.navigator.userAgent);
+
+    $rootScope.$on('dim-settings-loaded', function() {
+      var language = dimSettingsService.language;
+      if (chromeVersion && chromeVersion.length === 2 && parseInt(chromeVersion[1], 10) < 51) {
+        dimInfoService.show('old-chrome', {
+          title: $translate.instant('Help.UpgradeChrome'),
+          view: 'views/' + language + '/upgrade-chrome.html?v=$DIM_VERSION',
```

Comment:
the same way it does in [here](https://github.com/DestinyItemManager/DIM/blob/56970160aa4bea15625a511649aa1ab96c20c32b/src/scripts/shell/dimAppCtrl.controller.js#L137)